pkg/kubelet/images/image_manager.go

```diff
@@ -136,6 +138,19 @@ func (m *imageManager) EnsureImageExists(pod *v1.Pod, container *v1.Container, p
 			msg := fmt.Sprintf("image pull failed for %s because the registry is unavailable.", container.Image)
 			return "", msg, imagePullResult.err
 		}
+		// Pull image failure because image is corrupt: https://github.com/kubernetes/kubernetes/issues/46184
+		if strings.Contains(imagePullResult.err.Error(), "rename") && strings.Contains(imagePullResult.err.Error(), "directory not empty") {
```

Comment:
Since this is a moby specific issue, it should really be separated out into a moby specific location.  Also, why is this not being fixed in moby and an updated package required?  Generally we don't work around bugs in the container runtime like this.